mirrord/cli/src/operator/status.rs

```diff
@@ -0,0 +1,307 @@
+                    "Namespace",
+                    "Copy Pod Name",
+                    "Scale Down?"
+                ]);
+
+                for (pod_name, copy_target_resource) in copy_targets {
+                    copy_targets_table.add_row(row![
+                        copy_target_resource.spec.target.to_string(),
+                        copy_target_resource
+                            .metadata
+                            .namespace
+                            .as_deref()
+                            .unwrap_or_default(),
+                        pod_name,
+                        if copy_target_resource.spec.scale_down {
+                            "*"
+                        } else {
+                            ""
+                        },
+                    ]);
+                }
+
+                copy_targets_table.printstd();
+            }
+            println!();
+        }
+
+        if let Some(statistics) = status.statistics.as_ref() {
+            println!("Operator Daily Users: {}", statistics.dau);
+            println!("Operator Monthly Users: {}", statistics.mau);
+        }
+
+        let mut sessions = Table::new();
+
+        sessions.add_row(row![
+            "Session ID",
+            "Target",
+            "Namespace",
+            "User",
+            "Ports",
+            "Session Duration"
+        ]);
+
+        let mut sqs_rows: HashMap<QueueConsumer, Vec<Row>> = HashMap::new();
+
+        for session in &status.sessions {
+            let locked_ports = session
+                .locked_ports
+                .as_deref()
+                .map(|ports| {
+                    ports
+                        .iter()
+                        .map(|(port, type_, filter)| {
+                            format!(
+                                "Port: {port}, Type: {type_}{}",
+                                filter
+                                    .as_ref()
+                                    .map(|f| format!(", Filter: {}", f))
+                                    .unwrap_or_default()
+                            )
+                        })
+                        .collect::<Vec<_>>()
+                        .join("\n")
+                })
+                .unwrap_or_default();
+
+            sessions.add_row(row![
+                session.id.as_deref().unwrap_or_default(),
+                &session.target,
+                session.namespace.as_deref().unwrap_or("N/A"),
+                &session.user,
+                locked_ports,
+                humantime::format_duration(Duration::from_secs(session.duration_secs)),
+            ]);
+
+            if let Some(sqs_in_status) = session.sqs.as_ref().and_then(|sqs| {
+                Self::sqs_rows(
+                    sqs.iter(),
+                    session.id.clone().unwrap_or_default(),
+                    &session.user,
+                )
+            }) {
+                sqs_rows.extend(sqs_in_status);
```

Comment:
I think this overrides information from other sessions, doesn't it?
`sqs_rows` is a (consumer -> rows-vec) map, so if the first session has rows for consumer _a_ and also the second session has rows for consumer _a_, after handling the second session, `sqs_rows.get(a)` will only contain rows from the second session, because `extend` replaces existing keys that are specified in the extending map, right?

Reply:
Once again, you're right.

Now I think we're doing the proper merge.